Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc

```diff
@@ -1072,5 +1085,35 @@ void Phase2TrackerValidateDigi::fillHitsPerTrack() {
     nSimHitsPerTrack->Fill(lval, local_mes.nHits);
   }
 }
+*/
+std::string Phase2TrackerValidateDigi::getHistoId(uint32_t det_id, const TrackerTopology* tTopo, bool flag) {
+  int layer;
+  std::string Disc;
+  std::ostringstream fname1;
+  if(flag){
+    layer = tTopo->getITPixelLayerNumber(det_id);
+  }
+  else {
+    layer = tTopo->getOTLayerNumber(det_id);
+  }
+  if(layer < 0)
+    return "";
+
+  if (layer < 100) {
+    fname1 << "Barrel/";
+    fname1 << "Layer" << layer;
+    fname1 << "";
+  } else {
+    int side = layer / 100;
+    fname1 << "EndCap_Side" << side << "/";
+    int disc = layer - side * 100;
+    if (flag) Disc = (disc < 9) ? "FPIX_1" : "FPIX_2";
+    else Disc = (disc < 3) ? "TEDD_1" : "TEDD_2";
+    fname1 << Disc << "/";
+    int ring = tTopo->tidRing(det_id);
+    fname1 << "Ring" << ring;
```

Comment:
as above I am puzzled by how this works in the case of FPix.
Also this function is essentially a copy of the other one in `Phase2TrackerMonitorDigi.cc`. It would be cleaner to have some static method declared elsewhere and used in both classes.

Reply:
@mmusich This is in my todo list. It is also used in the other modules. I will move this out into a separate file after this PR is merged.